DataFormats/TrackerCommon/interface/TrackerTopology.h

```diff
@@ -600,6 +600,13 @@ class TrackerTopology {
   bool hasField(const DetId &id, DetIdFields idx) const {
     return id.subdetId() == bits_per_field[idx].subdet;
   }
+
+  const PixelBarrelValues& pbVals() const { return pbVals_; }
+  const PixelEndcapValues& pfVals() const { return pfVals_; }
+  const TOBValues& tobVals() const { return tobVals_; }
+  const TIBValues& tibVals() const { return tibVals_; }
+  const TIDValues& tidVals() const { return tidVals_; }
+  const TECValues& tecVals() const { return tecVals_; }
```

Comment:
@slava77 The difference wrt. #14586/#14230 is that here I'd like to not do the DetId parsing in CMSSW, but dump the constants and do the parsing afterwards (in PyROOT, in principle independently of CMSSW).

If this exposure of TrackerTopology internals is to be avoided at all cost, I can implement the detector generalization to the ntuple in another way (essentially adding a branch for each element the union of possible DetId elements per pixel and strip subdetectors (*), which doesn't look too nice but is the simplest alternative).

(*) e.g. for pixels: layer, ladder, module, side, disk, blade, panel